example/randomCells2D/src/randomCells2D.cpp

```diff
@@ -0,0 +1,187 @@
+    using Dim = alpaka::DimInt<2>;
+    using Idx = std::size_t;
+    using Vec = alpaka::Vec<Dim, Idx>;
+    using Acc = alpaka::ExampleDefaultAcc<Dim, Idx>;
+    using Host = alpaka::DevCpu;
+    auto const devAcc = alpaka::getDevByIdx<Acc>(0u);
+    auto const devHost = alpaka::getDevByIdx<Host>(0u);
+    using QueueProperty = alpaka::Blocking;
+    using QueueAcc = alpaka::Queue<Acc, QueueProperty>;
+    QueueAcc queue{devAcc};
+
+    using BufHost = alpaka::Buf<Host, float, Dim, Idx>;
+    using BufAcc = alpaka::Buf<Acc, float, Dim, Idx>;
+    using BufHostRand = alpaka::Buf<Host, alpaka::rand::Philox4x32x10<Acc>, Dim, Idx>;
+    using BufAccRand = alpaka::Buf<Acc, alpaka::rand::Philox4x32x10<Acc>, Dim, Idx>;
+    using BufHostRandVec = alpaka::Buf<Host, alpaka::rand::Philox4x32x10Vector<Acc>, Dim, Idx>;
+    using BufAccRandVec = alpaka::Buf<Acc, alpaka::rand::Philox4x32x10Vector<Acc>, Dim, Idx>;
+    using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+
+    constexpr Idx numX = NUM_X;
+    constexpr Idx numY = NUM_Y;
+
+    const Vec extent(numY, numX);
+
+    constexpr Idx perThreadX = 1;
+    constexpr Idx perThreadY = 1;
+
+    WorkDiv workdiv{alpaka::getValidWorkDiv<Acc>(
+        devAcc,
+        extent,
+        Vec(perThreadY, perThreadX),
+        false,
+        alpaka::GridBlockExtentSubDivRestrictions::Unrestricted)};
+
+    // Setup buffer.
+    BufHost bufHostA{alpaka::allocBuf<float, Idx>(devHost, extent)};
+    float* const ptrBufHostA{alpaka::getPtrNative(bufHostA)};
+    BufAcc bufAccA{alpaka::allocBuf<float, Idx>(devAcc, extent)};
+    float* const ptrBufAccA{alpaka::getPtrNative(bufAccA)};
+
+    BufHost bufHostB{alpaka::allocBuf<float, Idx>(devHost, extent)};
+    float* const ptrBufHostB{alpaka::getPtrNative(bufHostB)};
+    BufAcc bufAccB{alpaka::allocBuf<float, Idx>(devAcc, extent)};
+    float* const ptrBufAccB{alpaka::getPtrNative(bufAccB)};
+
+    BufHostRand bufHostRandA{alpaka::allocBuf<alpaka::rand::Philox4x32x10<Acc>, Idx>(devHost, extent)};
+    BufAccRand bufAccRandA{alpaka::allocBuf<alpaka::rand::Philox4x32x10<Acc>, Idx>(devAcc, extent)};
+    alpaka::rand::Philox4x32x10<Acc>* const ptrBufAccRandA{alpaka::getPtrNative(bufAccRandA)};
+
+    BufHostRandVec bufHostRandB{alpaka::allocBuf<alpaka::rand::Philox4x32x10Vector<Acc>, Idx>(devHost, extent)};
+    BufAccRandVec bufAccRandB{alpaka::allocBuf<alpaka::rand::Philox4x32x10Vector<Acc>, Idx>(devAcc, extent)};
+    alpaka::rand::Philox4x32x10Vector<Acc>* const ptrBufAccRandB{alpaka::getPtrNative(bufAccRandB)};
+
+    InitRandomKernel initRandomKernel;
+    alpaka::exec<Acc>(queue, workdiv, initRandomKernel, extent, ptrBufAccRandA);
+    alpaka::wait(queue);
+
+    alpaka::exec<Acc>(queue, workdiv, initRandomKernel, extent, ptrBufAccRandB);
+    alpaka::wait(queue);
+
+    for(Idx x = 0; x < numX; ++x)
+    {
+        for(Idx y = 0; y < numY; ++y)
+        {
+            ptrBufHostA[y * numX + x] = 0;
+            ptrBufHostB[y * numX + x] = 0;
+        }
+    }
```

Comment:
Is this something a `std::memset` or an `alpaka::memset` could do?

Reply:
Maybe. But `std::memset` assigns bytes so I'd have to calculate the correct extent. `alpaka::memset` takes queue as a parameter so I guess it's supposed to be used on the device side?

Reply:
Ah, I see. Because the buffer could have a pitched allocation. Ok fine!